diff -urN minetest-0.4.7/src/ban.h minetest-0.4.7-patched/src/ban.h
--- minetest-0.4.7/src/ban.h	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/ban.h	2013-06-09 20:30:42.645371651 +1100
@@ -26,6 +26,8 @@
 #include <jmutex.h>
 #include "exceptions.h"
 
+using namespace jthread;
+
 class BanManager
 {
 public:
diff -urN minetest-0.4.7/src/debug.h minetest-0.4.7-patched/src/debug.h
--- minetest-0.4.7/src/debug.h	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/debug.h	2013-06-09 20:30:42.645371651 +1100
@@ -31,6 +31,8 @@
 #include "exceptions.h"
 #include <map>
 
+using namespace jthread;
+
 #ifdef _WIN32
 	#define WIN32_LEAN_AND_MEAN
 	#ifndef _WIN32_WINNT
diff -urN minetest-0.4.7/src/emerge.h minetest-0.4.7-patched/src/emerge.h
--- minetest-0.4.7/src/emerge.h	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/emerge.h	2013-06-09 20:30:42.645371651 +1100
@@ -30,6 +30,31 @@
 	{ if (enable_mapgen_debug_info) \
 	infostream << "EmergeThread: " x << std::endl; }
 
+/* ------ from local jthread ------ */
+#include <semaphore.h>
+
+class Event {
+sem_t sem;
+
+public:
+Event() {
+sem_init(&sem, 0, 0);
+}
+
+~Event() {
+sem_destroy(&sem);
+}
+
+void wait() {
+sem_wait(&sem);
+}
+
+void signal() {
+sem_post(&sem);
+}
+};
+/* ------ from local jthread ------ */
+
 class Mapgen;
 struct MapgenParams;
 struct MapgenFactory;
diff -urN minetest-0.4.7/src/main.cpp minetest-0.4.7-patched/src/main.cpp
--- minetest-0.4.7/src/main.cpp	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/main.cpp	2013-06-09 20:30:42.647371676 +1100
@@ -79,6 +79,8 @@
 #include "sound.h"
 #include "sound_openal.h"
 
+using namespace jthread;
+
 /*
 	Settings.
 	These are loaded from the config file.
diff -urN minetest-0.4.7/src/profiler.h minetest-0.4.7-patched/src/profiler.h
--- minetest-0.4.7/src/profiler.h	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/profiler.h	2013-06-09 20:30:42.647371676 +1100
@@ -28,6 +28,8 @@
 #include "util/timetaker.h"
 #include "util/numeric.h" // paging()
 
+using namespace jthread;
+
 /*
 	Time profiler
 */
diff -urN minetest-0.4.7/src/quicktune.cpp minetest-0.4.7-patched/src/quicktune.cpp
--- minetest-0.4.7/src/quicktune.cpp	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/quicktune.cpp	2013-06-09 20:30:42.647371676 +1100
@@ -22,6 +22,8 @@
 #include <jmutexautolock.h>
 #include "util/string.h"
 
+using namespace jthread;
+
 std::string QuicktuneValue::getString()
 {
 	switch(type){
diff -urN minetest-0.4.7/src/server.cpp minetest-0.4.7-patched/src/server.cpp
--- minetest-0.4.7/src/server.cpp	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/server.cpp	2013-06-09 20:30:42.649371699 +1100
@@ -60,6 +60,8 @@
 #include "util/serialize.h"
 #include "defaultsettings.h"
 
+using namespace jthread;
+
 void * ServerThread::Thread()
 {
 	ThreadStarted();
diff -urN minetest-0.4.7/src/server.h minetest-0.4.7-patched/src/server.h
--- minetest-0.4.7/src/server.h	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/server.h	2013-06-09 20:32:34.502025571 +1100
@@ -43,6 +43,8 @@
 
 #define PP(x) "("<<(x).X<<","<<(x).Y<<","<<(x).Z<<")"
 
+using namespace jthread;
+
 class IWritableItemDefManager;
 class IWritableNodeDefManager;
 class IWritableCraftDefManager;
diff -urN minetest-0.4.7/src/settings.h minetest-0.4.7-patched/src/settings.h
--- minetest-0.4.7/src/settings.h	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/settings.h	2013-06-09 20:30:42.650371711 +1100
@@ -37,6 +37,8 @@
 #include <map>
 #include <set>
 
+using namespace jthread;
+
 enum ValueType
 {
 	VALUETYPE_STRING,
diff -urN minetest-0.4.7/src/tile.cpp minetest-0.4.7-patched/src/tile.cpp
--- minetest-0.4.7/src/tile.cpp	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/tile.cpp	2013-06-09 20:30:42.651371723 +1100
@@ -33,6 +33,8 @@
 #include "util/thread.h"
 #include "util/numeric.h"
 
+using namespace jthread;
+
 /*
 	A cache from texture name to texture path
 */
diff -urN minetest-0.4.7/src/util/container.h minetest-0.4.7-patched/src/util/container.h
--- minetest-0.4.7/src/util/container.h	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/util/container.h	2013-06-09 20:30:42.651371723 +1100
@@ -27,6 +27,8 @@
 #include <list>
 #include <vector>
 
+using namespace jthread;
+
 /*
 	Queue with unique values with fast checking of value existence
 */
diff -urN minetest-0.4.7/src/util/thread.h minetest-0.4.7-patched/src/util/thread.h
--- minetest-0.4.7/src/util/thread.h	2013-06-07 04:14:26.000000000 +1100
+++ minetest-0.4.7-patched/src/util/thread.h	2013-06-09 20:30:42.651371723 +1100
@@ -25,6 +25,8 @@
 #include <jmutex.h>
 #include <jmutexautolock.h>
 
+using namespace jthread;
+
 template<typename T>
 class MutexedVariable
 {
