diff -urN minetest-0.4.6/src/json/CMakeLists.txt minetest-0.4.6-patched/src/json/CMakeLists.txt
--- minetest-0.4.6/src/json/CMakeLists.txt	2013-04-16 21:34:46.395506506 +1100
+++ minetest-0.4.6-patched/src/json/CMakeLists.txt	2013-04-16 21:34:57.443526219 +1100
@@ -6,7 +6,7 @@
 	set(json_platform_LIBS "")
 endif( UNIX )
 
-add_library(jsoncpp ${json_SRCS})
+add_library(jsoncpp STATIC ${json_SRCS})
 
 target_link_libraries(
 	jsoncpp
